public/app/features/explore/Explore.tsx

```diff
@@ -111,12 +113,13 @@ export class Explore extends React.PureComponent<ExploreProps> {
   }
 
   componentDidMount() {
-    const { exploreId, urlState, initialized } = this.props;
-    const { datasource, queries, range = DEFAULT_RANGE, ui = DEFAULT_UI_STATE } = (urlState || {}) as ExploreUrlState;
+    const { exploreId, urlState, initialized, timeZone } = this.props;
+    const { datasource, queries, range: urlRange, ui = DEFAULT_UI_STATE } = (urlState || {}) as ExploreUrlState;
```

Comment:
Struggled a bit with this one. Seems to me it would have been more suitable if this logic was put in the initializeExplore action.  Then I could have just sent a DEFAULT_RANGE as initialRange and the action could have figured out if range should be taken from initialRange or urlstate. Thoughts?

I guess a general question here is where/how to handle default values since there's also DEFAULT_UI_STATE in use here in case of empty urlstate.

Reply:
Agree that those default values could/should be handled on action level. I don't like directly accessing store in components like this:  `const initialDatasource = datasource || store.get(LAST_USED_DATASOURCE_KEY);` especially

Reply:
Thanks :+1: Let's solve this in another refactor. Will try and create issue to track this

Reply:
Just noticed you left a lot of other comments. GitHub doesn't send notifications from time to time :cry:

Reply:
Moved it down to `mapStateToProps`